vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java

```diff
@@ -570,16 +569,11 @@ private void updateEnabled(RadioButton<T> button) {
         }
 
         button.setEnabled(!disabled);
-        Serializable rawValue = button.getElement().getPropertyRaw("disabled");
-        if (rawValue instanceof Boolean) {
-            // convert the boolean value to a String to force update the
-            // property value. Otherwise since the provided value is the same as
-            // the current one the update don't do anything.
-            button.getElement().setProperty("disabled",
-                    disabled ? Boolean.TRUE.toString() : null);
-        } else {
-            button.setDisabled(disabled);
-        }
+        button.setDisabled(disabled);
+        // When enabling a disabled radio group, individual button Web Components that should
+        // remain disabled (due to itemEnabledProvider), may end up rendering as enabled.
+        // Enforce the Web Component state using JS.
+        button.getElement().executeJs("this.disabled = $0", disabled);
```

Comment:
For some reason, simply setting the `disabled` property state for the button element isn't enough to get it to reflect properly on the Web Component (there's probably an issue in Flow enabled state handling). Added a JS call to enforce the state.

There were some earlier workarounds that aimed to tackle the same issue, removed them in favor of this approach.